WordPress/openverse-frontend
# Refactor Content report

## Fixes


Related to #543 by @panchovm 
Fixes #561 by @obulat

## Description


This PR removes the `report-content` Vuex store module, moving the functionality to the `VContentReportForm` component.

Other changes in this PR include:
- some styling adjustments to the report form (more work on this will be needed)
- renaming of the touched components to add the `V` prefix.
- refactoring `PhotoDetails` and all the components in `ContentReport` directory to use composition API.
- replacing Icon font icons with svg icons (`back`, `flag`).
- adding component imports to remove warnings in test output.
- rewriting the `photo-details.spec.js` tests to use `testing-library`.

## Testing Instructions

Run the app using `pnpm run dev`, open a single image result page, and click on 'Report'. The popup form should open with 3 options, and it should work correctly.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
